File: 793-preimage-size-of-factorial-zeroes-function/793-preimage-size-of-factorial-zeroes-function.py
class Solution:
    def preimageSizeFZF(self, k: int) -> int:
        def trailingZeros(n):
            res = n
            while n:
                n //= 5
                res += n
            return res

          
        l, r = 0, k + 1
        while l < r:
            mid = l + (r - l) // 2
            num = trailingZeros(mid)
            if num == k:
                return 5
            if num < k:
                l = mid + 1
            else:
                r = mid
        return 0

File: 793-preimage-size-of-factorial-zeroes-function/793-preimage-size-of-factorial-zeroes-function.js
/**
 * @param {number} k
 * @return {number}
 */
var preimageSizeFZF = function (k) {
  let [low, high] = [0, k + 1];

  while (low < high) {
    const mid = (low + high) >>> 1;
    const num = trailingZeros(mid);
    if (num === k) return 5;

    num < k ? (low = mid + 1) : (high = mid);
  }
  
  return 0;
};

const trailingZeros = (n) => {
  let res = n;

  while (n) {
    n = ~~(n / 5);
    res += n;
  }

  return res;
};
